src/composables/trade/useSor.ts

```diff
@@ -410,40 +393,35 @@ export default function useSor({
         tokenInDecimals,
         tokenOutDecimals,
         SwapTypes.SwapExactOut,
-        tokenOutAmount
+        tokenOutAmountScaled
       );
 
       sorReturn.value = swapReturn; // TO DO - is it needed?
 
-      const tradeAmount: BigNumber = swapReturn.returnAmount;
-      const tokenInAmountNormalised = bnum(
-        formatFixed(tradeAmount, tokenInDecimals)
-      );
-      tokenInAmountInput.value =
-        tokenInAmountNormalised.toNumber() > 0
-          ? formatAmount(tokenInAmountNormalised.toString())
-          : '';
+      let tokenInAmount = swapReturn.returnAmount;
+      tokenInAmountInput.value = tokenInAmount.gt(0)
+        ? formatAmount(formatUnits(tokenInAmount, tokenInDecimals))
+        : '';
 
       if (!sorReturn.value.hasSwaps) {
         priceImpact.value = 0;
       } else {
-        tokenOutAmountNormalised = await adjustedPiAmount(
-          tokenOutAmountNormalised,
-          tokenOutAddress,
-          tokenOutDecimals
-        );
+        tokenInAmount = await adjustedPiAmount(tokenInAmount, tokenOutAddress);
 
-        const effectivePrice = tokenInAmountNormalised.div(
-          tokenOutAmountNormalised
-        );
+        const divScale = BigNumber.from(10).pow(tokenInDecimals);
+        const wadScale = BigNumber.from(10).pow(18);
+        const effectivePrice = tokenOutAmountScaled
+          .mul(divScale)
+          .div(tokenInAmount);
         const priceImpactCalc = effectivePrice
-          .div(swapReturn.marketSpNormalised)
-          .minus(1);
+          .mul(wadScale)
+          .div(parseUnits(Number(swapReturn.marketSpNormalised).toFixed(18)))
+          .sub(ONE);
```

Comment:
`        const divScale = BigNumber.from(10).pow(tokenInDecimals);

        const wadScale = BigNumber.from(10).pow(18);
        const effectivePrice = tokenOutAmountScaled
          .mul(divScale)
          .div(tokenInAmount);
        const priceImpactCalc = effectivePrice
          .mul(wadScale)
          .div(parseUnits(Number(swapReturn.marketSpNormalised).toFixed(18)))
          .sub(ONE);`

seems this logic is duplicated above, may be put it in `calcPriceImpact()`?